src/lib/InstantSearch.js

```diff
@@ -426,32 +424,29 @@ See: https://www.algolia.com/doc/guides/building-search-ui/widgets/create-your-o
   }
 }
 
-export function enhanceConfiguration(searchParametersFromUrl) {
-  return (configuration, widgetDefinition) => {
-    if (!widgetDefinition.getConfiguration) return configuration;
-
-    // Get the relevant partial configuration asked by the widget
-    const partialConfiguration = widgetDefinition.getConfiguration(
-      configuration,
-      searchParametersFromUrl
-    );
+export function enhanceConfiguration(configuration, widgetDefinition) {
+  if (!widgetDefinition.getConfiguration) {
+    return configuration;
+  }
 
-    const customizer = (a, b) => {
-      // always create a unified array for facets refinements
-      if (Array.isArray(a)) {
-        return union(a, b);
-      }
+  // Get the relevant partial configuration asked by the widget
+  const partialConfiguration = widgetDefinition.getConfiguration(configuration);
 
-      // avoid mutating objects
-      if (isPlainObject(a)) {
-        return mergeWith({}, a, b, customizer);
-      }
+  const customizer = (a, b) => {
+    // always create a unified array for facets refinements
+    if (Array.isArray(a)) {
+      return union(a, b);
```

Comment:
```suggestion
      return [...a, ...b];
```

I think `b` could be not an array at this point, but it seems simpler (and less lodash)

Reply:
See https://github.com/algolia/instantsearch.js/pull/3632#discussion_r270817793.